### PR TITLE
For #9351 - Get camera image even with "Don't keep activities"

### PR DESCRIPTION
**Pull request by Mugurell**
The previous functionality was already based on saving the camera photo in a
specific file and then us reading it from there but what that file name was was
lost in the case of the activity being destroyed as being the case with "Don't
keep activities".
By making that file name top level it will now be available for the entire
duration of the app.

[video](https://drive.google.com/file/d/1kZKBa_bhiOJ7eXhHF3WTkGxaDaWb5CRS/view?usp=sharing)

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Review by Amejia481 — APPROVED** [review]
👍🏽

File: components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FilePicker.kt [hunk 1]
```diff
@@ -197,6 +198,8 @@ internal class FilePicker(
                 }
             } ?: request.onDismiss()
         }
+
+        captureUri = null
```
Comment (Amejia481) [inline components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FilePicker.kt:202]: Could we verify in tests that we are cleaning this value? :)

Comment (Mugurell) [reply components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FilePicker.kt:202]: Looked into this but I'm not sure how.
With `mockk` we would have access to this file scoped properties but I don't know if `mockito-kotlin` supports this.
Any pointers are much appreciated.

Comment (Amejia481) [reply components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FilePicker.kt:202]: As this property is static (global) I think we only need to change the visibility to `internal` and add the `@VisibleForTesting`, then verify from the test that after `handleFilePickerIntentResult` `captureUri` is `null`

Comment (Mugurell) [reply components/feature/prompts/src/main/java/mozilla/components/feature/prompts/file/FilePicker.kt:204]: Thank you!
